app/src/main/java/com/example/plantsapp/data/repository/RoomTasksRepository.kt

```diff
@@ -20,27 +21,31 @@ class RoomTasksRepository(
         )
     }
 
-    override suspend fun getTasksForDate(date: Date): Flow<List<Task>> {
+    override suspend fun getPlantsWithTasksForDate(date: Date): Flow<List<PlantWithTasks>> {
         return plantsWithTasksDao.getPlantsWithTasks()
             .map {
-                it.getAllFittingTasks(date)
+                it.getPlantsWithFittingTasks(date)
             }
     }
 
-    private fun List<RoomPlantWithTasks>.getAllFittingTasks(date: Date): List<Task> {
-        return this.flatMap {
-            it.tasks
-                .filter { roomTask ->
-                    checkIfDateIsRepeatedWithInterval(
-                        date,
-                        Date(it.plant.creationDateMillis),
-                        roomTask.frequency
-                    )
-                }
-                .map { roomTask ->
-                    roomTask.toTask()
-                }
+    private fun List<RoomPlantWithTasks>.getPlantsWithFittingTasks(date: Date): List<PlantWithTasks> {
+        return this.map {
+            PlantWithTasks(
+                it.plant.toPlant(),
+                it.tasks
+                    .filter { roomTask ->
+                        checkIfDateIsRepeatedWithInterval(
+                            date,
+                            Date(it.plant.creationDateMillis),
+                            roomTask.frequency
+                        )
+                    }
+                    .map { roomTask ->
+                        roomTask.toTask()
+                    }
+            )
```

Comment:
Extract filtering of tasks into reparate private function for better reading of this code

Reply:
Also if you want to, we can add some tests for these features. Just write to me, it you think that it will be cool (or we can proceed to other tasks and return to tests later)